internal/provider/utils.go

```diff
@@ -134,3 +134,13 @@ func formatEnumMarkdownList[T ~string](allowedValues []T) (mdList string) {
 	}
 	return mdList
 }
+
+type Knowable interface {
+	IsUnknown() bool
+	IsNull() bool
+}
+
+// IsKnown is a shortcut that checks in a value is neither null nor unknown.
+func IsKnown[T Knowable](t T) bool {
```

Comment:
nit: I'm curious if this needs the generic `[T Knowable]` or if `func IsKnown(v Knowable) bool` would suffice 🤷

Reply:
Nope, that's not necessary. I think I was originally planning on using a concrete base type but switched to an interface and forgot to adapt.